src/fonduer/parser/visual_linker.py

```diff
@@ -135,25 +132,36 @@ def _extract_pdf_words(self) -> None:
         if self.verbose:
             self.logger.info(f"Extracted {len(self.pdf_word_list)} pdf words")
 
-    def is_linkable(self, filename: str) -> bool:
-        """Verify that the file exists and has a PDF extension.
+    def _get_linked_pdf_path(self, filename: str, pdf_path: str = None) -> str:
+        """Get the linked pdf file path, return None if it doesn't exist.
 
-        :param filename: The path to the PDF document.
+        :param filename: The name to the PDF document.
+        :param pdf_path: The path to the PDF documents, if any, defaults to None.
         """
-        path = self.pdf_path
+        path = pdf_path if pdf_path is not None else self.pdf_path
         # If path is file, but not PDF.
         if os.path.isfile(path) and path.lower().endswith(".pdf"):
-            return True
+            return path
         else:
             full_path = os.path.join(path, filename)
             if os.path.isfile(full_path) and full_path.lower().endswith(".pdf"):
-                return True
-            elif os.path.isfile(os.path.join(path, filename + ".pdf")):
-                return True
-            elif os.path.isfile(os.path.join(path, filename + ".PDF")):
-                return True
+                return full_path
+            full_path = os.path.join(path, filename + ".pdf")
+            if os.path.isfile(full_path):
+                return full_path
+            full_path = os.path.join(path, filename + ".PDF")
+            if os.path.isfile(full_path):
+                return full_path
+
+        return None
 
-        return False
+    def is_linkable(self, filename: str, pdf_path: str = None) -> bool:
+        """Verify that the file exists and has a PDF extension.
+
+        :param filename: The path to the PDF document.
+        :param pdf_path: The path to the PDF documents, if any, defaults to None.
+        """
+        return False if self._get_linked_pdf_path(filename, pdf_path) is None else True
```

Comment:
I think `_get_linked_pdf_path` would better be merged into `is_linkable`.

Reply:
We use `_get_linked_pdf_path ` in two places. One in `link` func and another place is here. I propose to have a private func because I think it would be better to use the same func to check if the file is linkable and retrieve the real file path.

Reply:
I thought `_get_linked_pdf_path` was only used by `is_linkable`. Thanks for the clarification.